dattri/algorithm/influence_function.py

```diff
@@ -5,99 +5,154 @@
+            index (int): The index of the model parameters. This index
+                is used for ensembling of different trained model
+                parameters.
+            data (Tuple[Tensor]): The test data. Normally this is a tuple
+                of input data and target data, the number of items in the
+                tuple should be aligned in the target function. The tensors'
+                shape follows (1, batchsize, ...).
+
+        Returns:
+            torch.Tensor: The query based on the test data. Normally it is
+                a 2-d dimensional tensor with the shape of
+                (batchsize, num_parameters).
+        """
+        model_params, _ = self.task.get_param(index)
+        return self.task.get_grad_target_func()(model_params, data)
+
+    def generate_train_query(
+        self,
+        index: int,
+        data: Tuple[torch.Tensor, ...],
+    ) -> torch.Tensor:
+        """Calculating the query based on the train data.
+
+        Args:
+            index (int): The index of the model parameters. This index
+                is used for ensembling of different trained model
+                parameters.
+            data (Tuple[Tensor]): The train data. Normally this is a tuple
+                of input data and target data, the number of items in the
+                tuple should be aligned in the target function. The tensors'
+                shape follows (1, batchsize, ...).
+
+        Returns:
+            torch.Tensor: The query based on the train data. Normally it is
+                a 2-d dimensional tensor with the shape of
+                (batchsize, num_parameters).
+        """
+        model_params, _ = self.task.get_param(index)
+        return self.task.get_grad_target_func()(model_params, data)
+
+    @abstractmethod
+    def transformation_on_query(
+        self,
+        index: int,
+        data: Tuple[torch.Tensor, ...],
+        query: torch.Tensor,
+    ) -> torch.Tensor:
+        """Calculate the transformation on the query.
+
+        Inner product attributor calculates the inner product between the
+        test query and the transformation of the test query. This function
+        calculates the transformation of the test query based on the train data.
+
+        Normally speaking, this function may return any transformation on the query.
+        Hessian Matrix and Fisher Information Matrix are two common transformations.
+
+        Args:
+            index (int): The index of the model parameters. This index
+                is used for ensembling of different trained model.
+            data (Tuple[Tensor]): The train data. Normally this is a tuple
+                of input data and target data, the number of items in the
+                tuple should be aligned in the target function. The tensors'
+                shape follows (1, batchsize, ...).
+            query (torch.Tensor): The query based on the test data. Normally it is
```

Comment:
if this is always associated with test data, then just call it test_query?

Reply:
I will change the comment to "The query to be transformed."